### ISSUE TITLE
Gateway registration expires using fedimint-cli

### ISSUE THREAD

**Issue by douglaz** [fedimint/fedimint#4796]
I was able to reproduce this both on devimint and on production. After you join a federation, you can get:
```
$ fedimint-cli list-gateways
[
  {
(...)
    "vetted": false,
    "ttl": {
      "secs": 459,
      "nanos": 806087508
    }
  }
]
```
But after some minutes, the ttl will go to zero and never come back again:

```
$ fedimint-cli list-gateways
[
  {
(...)
    "vetted": false,
    "ttl": {
      "secs": 0,
      "nanos": 0
    }
  }
]
```

```
$ fedimint-cli --version
fedimint-cli 0.3.0-rc.3
```

This works as expected on fedimint-cli 0.2.2

**Comment by elsirion** [comment]
The client historically only updated gateways on demand when an LN invoice was to be generated or paid. Now that gateway selection in the CLI isn't done automatically anymore (to be fixed https://github.com/fedimint/fedimint/issues/4797#issuecomment-2031412587) the cache becomes stale after a while.

@maan2003 it will be interesting to see the impact of your gateway caching PR here, most of the time `fedimint-cli` might not run for long enough, but I think forced updates still happen when fetching gateways from an outdated cache.

**Comment by elsirion** [comment]
Closed by #4919
